source/simulator/assemblers/stokes.cc

```diff
@@ -125,38 +130,152 @@ namespace aspect
 
           const double JxW = scratch.finite_element_values.JxW(q);
 
-          if (this->get_parameters().use_full_A_block_preconditioner == false)
-            {
+          if(this->get_parameters().use_full_A_block_preconditioner==false){
+            for (unsigned int i = 0; i < stokes_dofs_per_cell; ++i)
+                  for (unsigned int j = 0; j < stokes_dofs_per_cell; ++j)
+                    if (scratch.dof_component_indices[i] ==
+                        scratch.dof_component_indices[j])
+                      {
+                        data.local_matrix(i, j) += ((2.0 * eta * (scratch.grads_phi_u[i]
+                                                                  * scratch.grads_phi_u[j]))
+                                                    )
+                                                  * JxW;
+                      }
+            
+
+          }
+          if(this->get_parameters().use_bfbt){
+            const double sqrt_eta=sqrt(eta);
+            const unsigned int pressure_component_index = this->introspection().component_indices.pressure;
+
               for (unsigned int i = 0; i < stokes_dofs_per_cell; ++i)
+                {
+                    for (unsigned int j = 0; j < stokes_dofs_per_cell; ++j)
+                      {
+
+
+                            // i and j are not pressures
+                          if(scratch.dof_component_indices[i]!=pressure_component_index && scratch.dof_component_indices[j]!=pressure_component_index)
+                            data.local_lumped_mass_matrix[i]+=sqrt_eta*scalar_product(scratch.phi_u[i],scratch.phi_u[j])*JxW;
+
+
+      // i and j are pressures?
+                          if(scratch.dof_component_indices[i]==pressure_component_index && scratch.dof_component_indices[j]==pressure_component_index)
+                          data.local_matrix(i, j) += (
+                                                      1.0/sqrt_eta * pressure_scaling
+                                                      * pressure_scaling
+                                                      * (scratch.grad_phi_p[i]
+                                                        * scratch.grad_phi_p[j] + 1e-6*scratch.phi_p[i]*scratch.phi_p[j] ))
+                                                    * JxW;
+                       }
+                                                      
+                          
+                          
+
+               }
+                
+
+          }
+          else{
```

Comment:
```suggestion
          else
          {
```
braces always go on separate lines in ASPECT.